src/utils/tests/numbers.test.ts

```diff
@@ -0,0 +1,3594 @@
+/* eslint-disable @typescript-eslint/no-loss-of-precision */
+import { Currency, CurrencyAmount, Fraction, Percent, Price, Token } from '@kyberswap/ks-sdk-core'
+import { afterAll, beforeAll, describe, expect, test, vi } from 'vitest'
+
+import { formatDisplayNumber } from '../numbers'
+
+describe('formatDisplayNumber tests', () => {
+  beforeAll(() => {
+    vi.mock('uuid', async importOriginal => {
+      const mod = (await importOriginal()) as any
+      return {
+        ...mod,
+        v4: () => '',
+      }
+    })
+  })
+  afterAll(() => {
+    vi.clearAllMocks()
+  })
+  describe('decimal', () => {
+    describe('number', () => {
+      describe('large numbers', () => {
+        describe('2 significantDigits', () => {
+          test('format number 1 correctly', async () => {
+            expect(formatDisplayNumber(1, { significantDigits: 2 })).toBe('1')
+          })
+          test('format number 12 correctly', async () => {
+            expect(formatDisplayNumber(12, { significantDigits: 2 })).toBe('12')
+          })
+          test('format number 123 correctly', async () => {
+            expect(formatDisplayNumber(123, { significantDigits: 2 })).toBe('120')
+          })
+          test('format number 1234 correctly', async () => {
+            expect(formatDisplayNumber(1234, { significantDigits: 2 })).toBe('1.2K')
+          })
+          test('format number 12345 correctly', async () => {
+            expect(formatDisplayNumber(12345, { significantDigits: 2 })).toBe('12K')
+          })
+          test('format number 123456 correctly', async () => {
+            expect(formatDisplayNumber(123456, { significantDigits: 2 })).toBe('120K')
+          })
+          test('format number 1234567 correctly', async () => {
+            expect(formatDisplayNumber(1234567, { significantDigits: 2 })).toBe('1.2M')
+          })
+          test('format number 12345678 correctly', async () => {
+            expect(formatDisplayNumber(12345678, { significantDigits: 2 })).toBe('12M')
+          })
+          test('format number 123456789 correctly', async () => {
+            expect(formatDisplayNumber(123456789, { significantDigits: 2 })).toBe('120M')
+          })
+          test('format number 123456789123456789 correctly', async () => {
+            expect(formatDisplayNumber(123456789123456789, { significantDigits: 2 })).toBe('120,000T')
+          })
+        })
+        describe('6 significantDigits', () => {
```

Comment:
why do we need these tests? also for 18? 
what is the special of 2 6 18? 
I think it is redundant.  Just 2 is enough

Reply:
We use 6 significantDigits in most cases. 2 is only for small space use. 18 significantDigits is to make sure it's rounding is correct